### ISSUE TITLE
Refunding Doesn't Update Credit Owed

### ISSUE THREAD

**Issue by gmacdougall** [solidusio/solidus#1475]
When a refund is given, the "Credit Owed" in the admin and order payment state are not correctly updated.
# Steps to Reproduce

Start up a Solidus Sandbox
1. Purchase 3 items
2. Go to the admin page for that order
3. Capture the payment for the order
4. Cancel one of the three items ordered
5. Go to the payment page and see that credit is owed
6. Refund the amount owed to the customer
## Expected Results

The order should move into a paid state and balance due/credit owed should be $0
## Actual Results

The order's payment state is still credit owed, with -$20.99 showing as credit owed at the top of the page.
# Example

![image](https://cloud.githubusercontent.com/assets/764390/18876922/7c43f292-8499-11e6-8632-6b09dd53f7cc.png)

**Comment by DanielePalombo** [comment]
My and @mtylty started to investigate this issue and we found these related issues: #1254, #1107, https://github.com/spree/spree/issues/6229

**Comment by skukx** [comment]
The reason this is happening is because of this line:
https://github.com/solidusio/solidus/blob/v2.1/core/app/models/spree/order.rb#L355
```ruby
    def outstanding_balance
      # If reimbursement has happened add it back to total to prevent balance_due payment state
      # See: https://github.com/spree/spree/issues/6229
      adjusted_payment_total = payment_total + refund_total

      if state == 'canceled'
        -1 * adjusted_payment_total
      else
        total - adjusted_payment_total
      end
    end
```

After a `Spree::Refund` is created `perform!` is called
https://github.com/solidusio/solidus/blob/master/core/app/models/spree/refund.rb#L16

In turn, this method calls `payment.order.updater.update` Which updates the `payment_total` column on the `Spree::Order`. This update already accounts for refunds when calculating `payment_total`.

See https://github.com/solidusio/solidus/blob/master/core/app/models/spree/order_updater.rb#L141

So TLDR; Get rid of `adjusted_payment_total` because `payment_total` is already adjusted.

**Comment by jhawthorn** [comment]
I believe this is fixed by #2002
